### PR TITLE
feat: add support for multiple grpc channels, fix configuration modifiers

### PR DESCRIPTION
**Pull request by cprice404**
This commit does the following:

* Fix the configuration modifier methods in the classes in the config tree; they were using `self.` methods, which are actually static methods
* Add support for configuring the number of grpc channels.

Tested the grpc channel support by modifying the example code and using `lsof`.

### CODE REVIEW

File: lib/momento/cache_client.rb [hunk 1]
```diff
@@ -339,9 +341,13 @@ def sorted_set_fetch_by_score(cache_name, sorted_set_name, min_score: nil, max_s
...
     def cache_stub
-      @cache_stub ||= CACHE_CLIENT_STUB_CLASS.new(@cache_endpoint, combined_credentials,
-        timeout: @configuration.transport_strategy.grpc_configuration.deadline
-      )
+      @cache_stubs ||= (1..@num_cache_stubs).map { CACHE_CLIENT_STUB_CLASS.new(@cache_endpoint, combined_credentials,
+        timeout: @configuration.transport_strategy.grpc_configuration.deadline,
+        channel_args: { 'grpc.use_local_subchannel_pool' => 1}
+      ) }
+      @next_cache_stub_index = (@next_cache_stub_index + 1) % @num_cache_stubs
+      puts "Next cache stub index: #{@next_cache_stub_index}"
```
Comment (nand4011) [inline lib/momento/cache_client.rb:349]: Is this print statement supposed to be here?

**Comment by cprice404** [comment]
@nand4011  This is rebased and the puts is removed.

File: lib/momento/cache_client.rb (hunk 1 above)
Comment (cprice404) [reply lib/momento/cache_client.rb]: nope thanks

**Comment by cprice404** [comment]
ugh looking into test failures and all this rubocop stuff

**Comment by cprice404** [comment]
okay @nand4011 this is green now :)

**Review by nand4011 — APPROVED** [review]

**Pull request merged**